cdap-app-fabric/src/main/java/co/cask/cdap/security/TokenSecureStoreUpdater.java

```diff
@@ -231,8 +238,36 @@ public long getUpdateInterval() {
 
   @Override
   public SecureStore update(String application, RunId runId) {
+    try {
+      if (Constants.Service.MASTER_SERVICES.equals(application)) {
+        // don't impersonate for the master.services Twill app
+        return doUpdate();
+      }
+
+      ProgramId programId = TwillAppNaming.fromTwillAppName(application);
```

Comment:
So we only really need the ``NamespaceId`` from the Twill app name, right? I think this is fine the way it is, I'm just making sure I follow.